### ISSUE TITLE
[HOLD for payment 2024-05-06] [HOLD #36071] [$1000] Textinput is not focused in chat after visiting conceirge help

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21401]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Create a new workspace
2. Go to the workspace invite page
3. Click `?` and then Get Assistance
4. Click chat with concierge
5. See that the textinput isn't focused (you have to manually click into it to begin typing)

## Expected Result: 
Text input should be focused
## Actual Result: 
Text input is not focused
(Also navigate other chats and their text input is not focused)

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.31-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
[Text input is not focused after visiting coneirge help.webm](https://github.com/Expensify/App/assets/43996225/fe3086e1-fe72-4497-806d-cb2351053026)

https://github.com/Expensify/App/assets/43996225/aa6f2326-5a98-4155-8187-8bc869d791a9

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Habtamu-Asefa
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686984617296819

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f42405b3f19296c4</li>
        <li>Upwork Job ID: 1674465220531503104</li>
        <li>Last Price Increase: 2023-07-06</li>
<li>Automatic offers: </li>
<ul>
<li>fedirjh | Reviewer | 0</li>
<li>bernhardoj | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @bfitzexpensify</details>

**Comment by StevenKKC** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Textinput is not focused in chat after visiting conceirge help.

### What is the root cause of that problem?
Compose is focused when below condition is true.
https://github.com/Expensify/App/blob/6183609c577f2310928bb4e8ee23b52f6dd28e00/src/pages/home/report/ReportActionCompose.js#L194

But after visiting conceirge help, `props.modal.isVisible`, i.e. `modal.isVisible` `onyx` data is not set to false, so compose is not focused.

### What changes do you think we should make in order to solve the problem?

We should set `modal.isVisible` `onyx` data set to true by calling `onCloseButtonPress` function after visiting conceirge help.
```
                        {this.props.shouldShowGetAssistanceButton && (
                            <Tooltip text={this.props.translate('getAssistancePage.questionMarkButtonTooltip')}>
                                <PressableWithoutFeedback
                                    onPress={() => {
                                        this.props.onCloseButtonPress();
                                        Navigation.navigate(ROUTES.getGetAssistanceRoute(this.props.guidesCallTaskID))
                                    }}
                                    style={[styles.touchableButtonImage]}
                                    accessibilityRole="button"
                                    accessibilityLabel={this.props.translate('getAssistancePage.questionMarkButtonTooltip')}
                                >
                                    <Icon src={Expensicons.QuestionMark} />
                                </PressableWithoutFeedback>
                            </Tooltip>
                        )}
```



### What alternative solutions did you explore? (Optional)
None.

**Comment by legendsoftdev** [comment]
Hi, thank you for your post.
I have worked on your problem and I hope I understood it correctly.
Please take a look at the attached file and let me know if you have any questions or feedback.

https://github.com/Expensify/App/assets/119021741/fbef49ee-d528-46dc-809d-dd669d287d51

Best regards.

**Comment by legendsoftdev** [comment]
Contributor details
Your Expensify account email: apexcup199096@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01d5e86a0f892dfc8f

**Comment by michaelhaxhiu** [comment]
- Adding "Hold" in title
- Linked to https://github.com/Expensify/App/issues/15992
- Switched to `Weekly` since it's blocked

**Comment by michaelhaxhiu** [comment]
@fedirjh do you think we can entertain proposals on this issue, or should I remove the `Help Wanted` label until the tracking GH is taken on?

I think we should decide based on whether or not we think it's possible to fix this independently from the tracking GH. Would be good to avoid a scenario in which we deploy a fix today and have to re-do that fix (or update that fix) in the future.

**Comment by michaelhaxhiu** [comment]
Chatted via DM with Fedi and we think this should be held. I removed the `Help Wanted` label.

**Comment by michaelhaxhiu** [comment]
still held

**Comment by michaelhaxhiu** [comment]
Note: I'm going OOO for ~2 weeks and making this `Monthly` because I don't foresee movement in the short term. If something changes, please remove my assignment & re-apply the Bug label to get a fresh BZ assigned to guide it forward.

**Comment by alphaboss1104** [comment]
Here is my [proposal](https://github.com/Expensify/App/issues/26353#issuecomment-1700256492).

**Comment by michaelhaxhiu** [comment]
I'm re-assigning this as part of my preparation for Sabbatical (starting tomorrow). In terms of **next steps**:

- This is held on https://github.com/Expensify/App/issues/15992
- Leaving as `Weekly` since it's blocked

**Comment by bfitzexpensify** [comment]
Stil held on https://github.com/Expensify/App/issues/15992. Since that isn't assigned to anyone to work on, I'm going to move this to monthly.

**Comment by bfitzexpensify** [comment]
Still on hold for https://github.com/Expensify/App/issues/15992

**Comment by bfitzexpensify** [comment]
Tracking issue closed, let's get this one retested.

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests.

**Comment by bfitzexpensify** [comment]
Nice, let's close this out then

**Comment by Habtamu-Asefa** [comment]
I reported this bug, will there be a bonus?

**Comment by bernhardoj** [comment]
@bfitzexpensify hi, the issue is still happening. QA also says they [can reproduce it](https://github.com/Expensify/App/issues/21401#issuecomment-1966990285).

I just updated my proposal based on the latest code and propose to fix this on some pages too where this issue is happening.

**Comment by bfitzexpensify** [comment]
Reopening since I misread https://github.com/Expensify/App/issues/21401#issuecomment-1966990285 😅

**Comment by bfitzexpensify** [comment]
OK, given it's happening on multiple pages as pointed out by @bernhardoj, I think this is something we should address

**Comment by bfitzexpensify** [comment]
@fedirjh looks like we have a couple of old proposals, and an updated one from @bernhardoj - do any of those work?

**Comment by bernhardoj** [comment]
Yes, the testing step is a bit outdated. You can find the Get Assistance in workspace invite page.
<img width="392" alt="image" src="https://github.com/Expensify/App/assets/50919443/a172838f-4810-4d7f-8aa4-ac16e68ad542">

**Comment by bfitzexpensify** [comment]
Thanks @bernhardoj - updated the testing steps in the OP

**Comment by bernhardoj** [comment]
> specifically using the browser's back button will not work as expected anymore.

Do you mean going back won't go back to the get assistance page? If yes, then that's the expected behavior. We want to close the RHP. You can test it with other RHP flows such as request money (for [reference](https://expensify.slack.com/archives/C01GTK53T8Q/p1686845966376029?thread_ts=1686126333.958959&cid=C01GTK53T8Q)).

**Comment by bernhardoj** [comment]
If you log the auto-focus value here,
https://github.com/Expensify/App/blob/faff990f6ce64fbef53712169b23788e3fd59ad0/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx#L278-L279

the value will be `false` because the `modal.isVisible` value is stuck at true.

I found out that after we use live markdown (RNMarkdownTextInput) on the web, setting the autofocus prop to false doesn't work. It always auto-focus the input and looks like that happens because we are setting the selection here.
https://github.com/Expensify/react-native-live-markdown/blob/7c9d9b050874b8b2a0ef8d000304b428dc8ec0e8/src/MarkdownTextInput.web.tsx#L527-L533

Commenting the effect solves the auto-focus issue.

cc: @Skalakid @BartoszGrajdek I guess

**Comment by bfitzexpensify** [comment]
OK, so are we good to close this out as no longer an issue?

**Comment by bernhardoj** [comment]
We are currently blocked by a live markdown bug on the web, I have [posted](https://github.com/Expensify/App/issues/36071#issuecomment-2031031366) it on the tracking issue.

**Comment by bfitzexpensify** [comment]
Held on https://github.com/Expensify/App/issues/36071

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bernhardoj** [comment]
@fedirjh I don't have an answer to that. I thought they are going to handle it because @tomekzaw +1 my report (😂), but I don't see it's being added to the list. @tomekzaw can you share any plan for that issue specifically, if any? Or should I create an issue on the `react-native-live-markdown` repo?

**Comment by Skalakid** [comment]
@bernhardoj thanks for reporting the issue, I investigate it and come back to you with more details

**Comment by bfitzexpensify** [comment]
Thanks @Skalakid

**Comment by Skalakid** [comment]
@bernhardoj The [PR](https://github.com/Expensify/react-native-live-markdown/pull/319) with the fix for the [issue](https://github.com/Expensify/App/issues/36071#issuecomment-2031031366) you reported has been just merged

**Comment by bernhardoj** [comment]
@Skalakid thanks a lot!

I can confirm the text input issue is resolved.

I think we can update `react-native-live-markdown` and fix this issue in one PR.

@fedirjh

**Comment by bernhardoj** [comment]
@Skalakid yes, that would be great

**Comment by Skalakid** [comment]
Hmm after bumping the `react-native-live-markdown` version the issue is reproducible again. To fix it we should apply @bernhardoj's proposal changes. Maybe let's bump the version of the package to `0.1.63` and add proposed changes in one PR? What do you think about it?

**Comment by bernhardoj** [comment]
@Skalakid that works. I can do it after @fedirjh approves and I get assigned to this issue 👍

**Comment by bernhardoj** [comment]
PR is ready

cc: @fedirjh

**Comment by Habtamu-Asefa** [comment]
Reporting bonus payment is pending. @bfitzexpensify the issue was created while bug reporting was being compensated $250

**Comment by bfitzexpensify** [comment]
Payment complete to @bernhardoj 

Offer sent @Habtamu-Asefa 

Please complete the BZ checklist @fedirjh and I'll finalise your payment.

**Comment by Habtamu-Asefa** [comment]
@bfitzexpensify Thank you, I just accepted the offer.

**Comment by bfitzexpensify** [comment]
Thanks @fedirjh. All payments complete, closing this out.
